### PR TITLE
Add support for encrypting PSKs in Mongo DB based registry

### LINKED ISSUE #2717: Add possibility for encrypting PSK keys in device registry
In the (MongoDB) device registry all of the credential types except for PSKs are hashed for security reasons.
PSKs which are used for CoAP devices however cannot be hashed because of their very nature.
To store such secrets more securely I propose to add the possibility for encrypting these secrets.

This could be achieved by providing a configurable crypt key which is used for encrypting and decrypting the secrets inside the device registry.

### PR DESCRIPTION
**Pull request by sophokles73**
The Mongo DB based registry has been extended to (transparently)
encrypt/decrypt pre-shared keys when writing/reading credentials to/from
the Mongo DB collection.

Fixes #2717

### CODE REVIEW

**Comment by sophokles73** [comment]
@calohmn @mbaeuerle would you mind taking a look?

File: services/device-registry-base/src/test/java/org/eclipse/hono/service/management/credentials/PskSecretTest.java [hunk 1]
```diff
@@ -79,4 +83,49 @@ public void testMergePropertiesUsesExistingKey() {
 
         assertThat(updatedSecret.getKey()).isEqualTo(new byte[] { 0x01, 0x02 });
     }
+
+    /**
+     * Verifies that the shared key is being encrypted/decrpyted using the given field level encryption helper.
```
Comment (calohmn) [inline services/device-registry-base/src/test/java/org/eclipse/hono/service/management/credentials/PskSecretTest.java:88]: Typo (decrpyted)

File: services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/util/CryptVaultBasedFieldLevelEncryption.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public byte[] encrypt(final byte[] data) {
+        return cryptVault.encrypt(data);
+    }
+
+    @Override
+    public byte[] decrypt(final byte[] data) {
+        return cryptVault.decrypt(data);
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/util/CryptVaultBasedFieldLevelEncryption.java:45]: nice catch :+1:

File: services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/model/MongoDbBasedCredentialsDao.java [hunk 3]
```diff
@@ -172,7 +164,7 @@ public void registerLivenessChecks(final HealthCheckHandler livenessHandler) {
                 .start();
 
         final Promise<String> addCredentialsPromise = Promise.promise();
-
+        credentials.getCredentials().parallelStream().forEach(cred -> cred.encryptFields(fieldLevelEncryption));
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/model/MongoDbBasedCredentialsDao.java:167]: I do not really mind. The crypt operations should be fast as we are talking about AES being used and the cipher/plain texts are small. I will simply change it to `stream()` ...

**Comment by sophokles73** [comment]
@calohmn I have pushed changes

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**